### PR TITLE
Testing the encoding.

### PR DESCRIPTION
**Pull request by lemire**
This adds more tests... this time on the encoding.

### CODE REVIEW

**Comment by lemire** [comment]
Merging since this only adds new tests.

File: tests/wpt_tests.cpp [hunk 1]
```diff
@@ -172,7 +187,7 @@ bool setters_tests_encoding() {
 
         // TODO: Handle invalid utf-8 tests too.
         if (!element["expected"]["hostname"].get(expected)) {
-          ada::set_host(base, std::string{new_value});
+          ada::set_host(base, std::string{new_value}, type);
```
Comment (anonrig) [inline tests/wpt_tests.cpp:190]: I think encoding type should not be a parameter and it should use base’s encoding type.

File: tests/wpt_tests.cpp [hunk 2]
```diff
@@ -154,7 +169,7 @@ bool setters_tests_encoding() {
 
       if (category == "protocol") {
         std::string_view expected = element["expected"]["protocol"];
-        ada::set_scheme(base, std::string{new_value});
+        ada::set_scheme(base, std::string{new_value}, type);
```
Comment (anonrig) [inline tests/wpt_tests.cpp:172]: Same applies to here regarding the encoding type parameter.

File: tests/wpt_tests.cpp [hunk 3]
```diff
@@ -146,6 +147,20 @@ bool setters_tests_encoding() {
       if (!element["comment"].get(comment)) {
         std::cout << "    comment: " << comment << std::endl;
       }
+      std::string_view encoding{};
+      ada::encoding_type type = ada::encoding_type::UTF8;
+      if (!element["encoding"].get(encoding)) {
+        std::cout << "    encoding: " << encoding << std::endl;
+        if(encoding == "UTF-8") {
```
Comment (anonrig) [inline tests/wpt_tests.cpp:154]: I think we should have a to_encoding(std::string_view) function for this

File: tests/wpt_tests.cpp (hunk 1 above)
Comment (lemire) [reply tests/wpt_tests.cpp:190]: It is part of the API right now. 

https://github.com/ada-url/ada/blob/f0298674750eeae848c5859aff062e91ef248e4c/include/ada/implementation.h#L67


Either the parameter should not appear in the public-facing API or else we must test it.

File: tests/wpt_tests.cpp (hunk 2 above)
Comment (lemire) [reply tests/wpt_tests.cpp:172]: It is a parameter in the API: https://github.com/ada-url/ada/blob/f0298674750eeae848c5859aff062e91ef248e4c/include/ada/implementation.h#L44

We must either remove it from the API or include tests for it.

File: tests/wpt_tests.cpp (hunk 3 above)
Comment (lemire) [reply tests/wpt_tests.cpp:154]: Agreed.

**Pull request merged**